internal/lint/tools.go

```diff
@@ -0,0 +1,14 @@
+// Conventionally this sort of thing would use the "ignore" tag, but
+// `go mod tidy` ignores so-tagged files explicitly, so we use another build
+// tag we never intend to set.
+// +build tools
+//go:build tools
```

Comment:
So is `go:build` in Go 1.17? I remembered something about "finishing the transition in Go 1.18" (which maybe was just dropping `// +build` support)?

I also recall that some old versions of Go required `// +build tools` to be the *very first line* of the file... but that might have been before `go mod` so maybe not an issue? 🤷‍♂️ I think the current rules (until 1.18) are:

1. start with the prefix +build followed by one or more spaces
2. located at top of file *before* the package declaration
3. have at least one blank line between it and package declaration to prevent it from being treated as package documentation
4. The tags separated by space will be interpreted under the OR logic.
5. Comma separated tags will be interpreted under the AND logic.
6. Each term is an alphanumeric word and if preceded by ! it means it is negated.

Reply:
Yeah, I think 1.16-1.18 all sync the two, and 1.17 is where `go:build` becomes the source of truth.  Since it's easy, I plan to keep both around until we have other reasons to drop 1.16, which is presumably a while yet.

With all that said, I could probably remove this build tag now; nobody should be trying to build anything in this module!